clients/gateway.go

```diff
@@ -44,7 +76,16 @@ func (c *GatewayClient) buildQueryString(endpoint string, args map[string]string
 
 // get performs a "GET" http request with the given URL and returns the response body
 func (c *GatewayClient) get(queryUrl string) ([]byte, error) {
```

Comment:
We probably need `get` to be cancelable as well, if not we might have to wait 34 mins just to get back the control of execution when we make a request. Can you see if we can use https://pkg.go.dev/net/http#NewRequestWithContext ?

Reply:
Good point. Since will require modifying a lot of code, it will be handled in a separate PR.